src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.Android.cs

```diff
@@ -0,0 +1,391 @@
+        {
+            value = null;
+            e = null;
+
+            value = id == LocalId ? GetLocalTimeZoneCore() : GetTimeZone(id, id);
+
+            if (value == null)
+            {
+                e = new InvalidTimeZoneException(SR.Format(SR.InvalidTimeZone_InvalidFileData, id, GetTimeZoneDirectory() + TimeZoneFileName));
+                return TimeZoneInfoResult.TimeZoneNotFoundException;
+            }
+
+            return TimeZoneInfoResult.Success;
+        }
+
+        private static string[] GetTimeZoneIds()
+        {
+            return AndroidTzDataInstance.GetTimeZoneIds();
+        }
+
+        /*
+        * Android v4.3 Timezone support infrastructure.
+        *
+        * This is a C# port of libcore.util.ZoneInfoDB:
+        *
+        *    https://android.googlesource.com/platform/libcore/+/master/luni/src/main/java/libcore/util/ZoneInfoDB.java
+        *
+        * This is needed in order to read Android v4.3 tzdata files.
+        *
+        * Android 10+ moved the up-to-date tzdata location to a module updatable via the Google Play Store and the
+        * database location changed (https://source.android.com/devices/architecture/modular-system/runtime#time-zone-data-interactions)
+        * The older locations still exist (at least the `/system/usr/share/zoneinfo` one) but they won't be updated.
+        */
+        private sealed class AndroidTzData
+        {
+            [StructLayout(LayoutKind.Sequential, Pack=1)]
+            private unsafe struct AndroidTzDataHeader
+            {
+                public fixed byte signature[12]; // "tzdata2012f\0"
+                public int indexOffset;
+                public int dataOffset;
+                public int finalOffset;
+            }
+
+            [StructLayout(LayoutKind.Sequential, Pack=1)]
+            private unsafe struct AndroidTzDataEntry
+            {
+                public fixed byte id[40];
+                public int byteOffset;
+                public int length;
+                public int unused; // Was raw GMT offset; always 0 since tzdata2014f (L).
+            }
+
+            private string[] _ids;
+            private int[] _byteOffsets;
+            private int[] _lengths;
+
+            public AndroidTzData()
+            {
+                string tzFilePath = GetTimeZoneDirectory() + TimeZoneFileName;
+                using (FileStream fs = File.OpenRead(tzFilePath))
+                {
+                    ReadHeader(tzFilePath, fs);
+                }
+            }
+
+            [MemberNotNull(nameof(_ids))]
+            [MemberNotNull(nameof(_byteOffsets))]
+            [MemberNotNull(nameof(_lengths))]
+            private unsafe void ReadHeader(string tzFilePath, Stream fs)
+            {
+                int size = Math.Max(Marshal.SizeOf(typeof(AndroidTzDataHeader)), Marshal.SizeOf(typeof(AndroidTzDataEntry)));
```

Comment:
```suggestion
                int size = Math.Max(sizeof(AndroidTzDataHeader), sizeof(AndroidTzDataEntry));
```
All places that use `Marshal.SizeOf` in this file should be replaced by `sizeof`. `sizeof` is at least order of magnitude more performant compared to `Marshal.SizeOf`.

Reply:
Thank you!